docs/zkapps/tutorials/11-advanced-account-updates.mdx

```diff
@@ -0,0 +1,169 @@
+
+Under the hood, what is happening is actually a bit more general than that. The Manager Account for a token actually controls all properties of token accounts.
+
+## zkApp Manager Accounts & Tokens
+
+For an Account Update to be able to modify state (including sending tokens) on a token account, two things must be true:
+
+1. The account update must meet the criteria of the permissions on the zkApp account itself. If the zkApp Account is setup with the `proof` permission, this means the Account Update proof must pass the verification key in the account
+2. The account update must set `mayUseToken` to True. An account update on a token account cannot set this property to true itself however. It must get its account update “approved” by its corresponding manager account. The manager account can approve accounts based on its `access` permission.
+
+## zkApp Token Management
+
+As an example, you can write a token that can approve a zkApp account to send tokens from itself to another zkApp.
+
+To implement this, you would have 3 contracts:
+
+1. `MyToken`: The token contract. On instantiation, serves as the zkAppManagerContract for all instances of the token.
+2. `TokenUser`: A zkApp contract which will manage tokens. This one is instantiated on the Mina (default) tokenId
+3. `TokenHolder`: A zkApp contract which will store tokens on behalf of the `TokenUser` contract. This one is instantiated on the `MyToken` instance TokenId.
+
+You can find the full code [here](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/11-advanced-account-updates/).
+
+A token standard library is in the above which will abstract much of these details away too for standard use. The illustration here though should help explain what that library is doing and how to think about zkAppManagerAccounts in detail.
+
+These contracts can be setup as follows:
+
+```typescript
+  const myTokenInstance = new MyToken(myTokenAddr);
+  const tokenUserInstance = new TokenUser(tokenUserAddr);
+  const tokenHolderInstance = new TokenHolder(tokenUserAddr, myTokenInstance.token.id);
+```
+
+And deployed as follows:
+
+```typescript
+  const deploy_txn = await Mina.transaction(deployerAccount, () => {
+    let feePayerUpdate = AccountUpdate.fundNewAccount(deployerAccount, 4);
+    feePayerUpdate.send({ to: myTokenAddr, amount: accountFee });
+
+    myTokenInstance.deploy();
+    tokenUserInstance.deploy();
+    tokenHolderInstance.deploy()
+
+    myTokenInstance.approveDeploy(tokenHolderInstance.self);
+  });
+```
+
+The special thing to note here is the `approveDeploy(...)` on the `myTokenInstance`. Because `tokenHolderInstance` is going to exist as a token account, it needs approval to be deployed.
+
+The `myTokenInstance.approveDeploy(...)` code is as follows:
+
+```typescript
+  @method approveDeploy(deployUpdate: AccountUpdate) {
+    // TODO should anything else be asserted here for a "safe" deployment?
+    this.approve(deployUpdate, AccountUpdate.Layout.NoChildren);
```

Comment:
```suggestion
```typescript
  @method approveDeploy(deployUpdate: AccountUpdate) {
    this.approve(deployUpdate, AccountUpdate.Layout.NoChildren);
```
I think you should delete the TODO comment :D

The question of whether an account update is "safe" depends on the application.

For example, a manager contract for NFTs could store NFTs in the 8-field state of each user account. In that case, being safe means that users cannot arbitrarily change their 8-field state. So, this method here would check that the account update doesn't modify state.

In the case of a simple token contract, the only aspect it wants to control about user's accounts is the token balance. So it needs to check that the token balance behaves according to its rules. Since your methods does that, it's safe.

Reply:
Maybe let's say, "// NOTE you should include any other application-specific constraints you would like here, for example asserting that an account cannot arbitrarily change its 8-field state"?

Reply:
Good idea, I'll include something to that effect in the code and maybe also in the tutorial text